### PR TITLE
Feature/dom based insert mode

### LINKED ISSUE #829: Canvas Selection: make insert mode use the dom-based selection

### PR DESCRIPTION
**Pull request by balazsbajorics**
Fixes #829 

**Problem:**
The insert mode was still using the old hittarget divs.

**Fix:**
Use the dom-based highlight functionality of the select-mode-hooks.

**Commit Details:**
- I kept most of the actual logic in insert-mode-control-container to avoid a massive PR
- I refactored the highlight hook so it can take a getter function for potential highlight targets. implemented two functions (`gethHiglightableViewsForInsertMode` for insert mode, and `getSelectableViewsForSelectMode` for select mode). the two functions are wrapped in hooks so they have access to ref editor state.

### CODE REVIEW

File: editor/src/components/canvas/controls/insert-mode-control-container.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-unused-expressions */
```
Comment (Rheeseyb) [inline editor/src/components/canvas/controls/insert-mode-control-container.tsx:1]: What is this being used for? Please can we only use this around the actual expressions we want to ignore.

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/controls/select-mode/insert-mode-hooks.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  const storeRef = useRefEditorState((store) => {
+    return {
+      componentMetadata: store.editor.jsxMetadataKILLME,
+      mode: store.editor.mode,
+      imports: getOpenImportsFromState(store.editor),
+    }
+  })
+  return React.useCallback(
+    // we don't use allElementsDirectlySelectable and childrenSelectable in insert mode, but they are needed for select mode
+    (_allElementsDirectlySelectable: boolean, _childrenSelectable: boolean) => {
```
Comment (seanparsons) [inline editor/src/components/canvas/controls/select-mode/insert-mode-hooks.ts:25]: Surely just don't include the parameters if none of them are used?

File: editor/src/components/canvas/controls/insert-mode-control-container.tsx (hunk 1 above)
Comment (balazsbajorics) [reply editor/src/components/canvas/controls/insert-mode-control-container.tsx:1]: eslint has a problem if you call a function that involves `?.` the no-unused-expressions rule is useful to catch the mistake of a line of code that does nothing, but I wonder if I should just disable this rule, given that we don't usually make that kind of mistake.

alternatively we should bump eslint, but that involves changing the eslint we embed in the editor. or we should have two eslint dependencies, one aliased for bundling, and one for our own actual use.

File: editor/src/components/canvas/controls/select-mode/insert-mode-hooks.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import {
+  insertionSubjectIsDragAndDrop,
+  insertionSubjectIsJSXElement,
+  isInsertMode,
+} from '../../../editor/editor-modes'
+import { getOpenImportsFromState } from '../../../editor/store/editor-state'
+import { useRefEditorState } from '../../../editor/store/store-hook'
+import { useHighlightCallbacks } from './select-mode-hooks'
+
+function useGethHiglightableViewsForInsertMode() {
```
Comment (seanparsons) [inline editor/src/components/canvas/controls/select-mode/insert-mode-hooks.ts:15]: `Geth`? :)

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/controls/select-mode/insert-mode-hooks.ts (hunk 3 above)
Comment (balazsbajorics) [reply editor/src/components/canvas/controls/select-mode/insert-mode-hooks.ts:15]: geth!

**Pull request merged**